plugins/scaffolder-backend-module-gitlab/src/actions/createProjectAccessToken.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
+import { ScmIntegrationRegistry } from '@backstage/integration';
+import { getToken } from '../util';
+
+/**
+ * Creates a `gitlab:create-project-access-token` Scaffolder action.
+ *
+ * @param options - Templating configuration.
+ * @public
+ */
+export const createGitlabProjectAccessToken = (options: {
+  integrations: ScmIntegrationRegistry;
+}) => {
+  const { integrations } = options;
+  return createTemplateAction<{
+    repoUrl: string;
+    projectId: string | number;
+    name: string;
+    accessLevel: number;
+    scopes: string[];
+    token?: string;
+  }>({
+    id: 'gitlab:create-project-access-token',
+    schema: {
+      input: {
+        required: ['projectId', 'repoUrl'],
+        type: 'object',
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            type: 'string',
+          },
+          projectId: {
+            title: 'Project ID',
+            type: 'string | number',
```

Comment:
I think you define this like `['string', 'number']` instead.

You can also [take advantage of `zod` schemas](https://backstage.io/docs/features/software-templates/writing-custom-actions#writing-your-custom-action) if you would like to instead of handcrafting both the jsonschema and the type generics. Makes all the actions a little simpler.